src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -279,9 +280,11 @@ public bool TryGetTemplateFromConfigInfo(IFileSystemInfo templateFileConfig, out
 
                 // Record the timestamp of the template file so we
                 // know to reload it if it changes
-                if (host.FileSystem is IFileLastWriteTimeSource timeSource)
+                if (templateFile.MountPoint.Info.MountPointFactoryId == FileSystemMountPointFactoryId &&
+                    host.FileSystem is IFileLastWriteTimeSource timeSource)
                 {
-                    runnableProjectTemplate.ConfigTimestampUtc = timeSource.GetLastWriteTimeUtc(templateFile.FullPath);
+                    var physicalPath = Path.Combine(templateFile.MountPoint.Info.Place, templateFile.FullPath.TrimStart('/'));
```

Comment:
This looks a bit as a workaround. Should we fix `FullPath` to contain full path to the file instead and not the relative path to mount point?  Or rename property?

Reply:
I'm not fan of it either... As I mentioned in commit message, we should probably add this API to `IFile`... See same hack at https://github.com/dotnet/templating/blob/3be78f6/src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs#L295-L301

Reply:
For example... FileSystem can't go inside .zip file and check this... but IFile that represents ZipFile entry could handle that....

Reply:
It is a problem only of this specific case? Or `FullPath` in general contains the relative path to mount point?

----
If it's an overall issue:

Modifying `Abstractions` is tricky as it is public and known to be used and we tend to avoid doing the changes.
I would tend to rename `FullPath` to reflect the meaning however it is a breaking change. Probably adding the new property can be an option.

Also, we can consider extension method that returns physical path (especially if its known to be used not once - probably there might be more cases).

Reply:
While working on fixing this I noticed that this pattern of combining .Place with FullPath.TrimStart("/") is pretty common for PhysicalFileSystem: see https://github.com/dotnet/templating/blob/master/src/Microsoft.TemplateEngine.Edge/Mount/FileSystem/FileSystemMountPoint.cs
So I suggest we merge this PR and work on new PR that fixes API around `LastWriteTimeUtc`, does that sound reasonable?